### DISCUSSION TITLE
Library: verde, function: rolling_window

### DISCUSSION THREAD

**Discussion by jgsnow** [fatiando/community#155] [Q&A] [answered]
Hello,

I intend to use the verde function called "rolling_window", and I have the following two questions.

1. I was wondering if it is possible to apply this function to an un-gridded data (xyz file) that contains 3 columns: easting, northing, anomaly values. I was not able to comprehend it completely from the documentation (https://www.fatiando.org/verde/latest/api/generated/verde.rolling_window.html). 

2. I could not quite understand the 2 essential parameters - size (float), shape(float). Both says "window size". For "size", the documentaion says 'The SIZE of the WINDOWS.' and for "shape", it says 'The WINDOW SIZE in the South-North and West-East directions ....'. It is confusing to me.

Thank you very much, in advance!

Jay

**Comment by santisoler** [comment, marked as answer]
Hi @jgsnow! Thanks for opening this topic. Trying to reply to your questions here:

1. Yes. You can use the rolling window function to scatter points. That's the point of this function. Applying rolling windows to gridded data can be done much more efficiently than what the rolling window implements (kd-trees), since with gridded data you can rely on indices for selecting points. You're right about the examples in the docs: they only use gridded coordinates. We should change them to use scattered points as well.
2. The description for `spacing` is wrong. The `spacing` is the distance between the centers of neighboring windows. We should update it as well. A better description of it is given in the last paragraph of the docstring, before listing the parameters.

Thanks again for reporting these points! We'll try to fix them soon.

**Comment by santisoler** [comment]
Opened https://github.com/fatiando/verde/pull/458 to address the issues mentioned here.
